### DISCUSSION TITLE
[V5] get the CTX types

### DISCUSSION THREAD

**Discussion by c4n4r** [adonisjs/core#2864] [Help] [answered]
I'm discovering the framework, and I ask to myself...
Is it possible to get the types of : Request, Auth, Response... Everything inside the CTX ?

I'd like to use Typescript at it's maximum and it could be cool to get this.

```
`async process({**request, auth**}) {

    const body = request.body();
    const token = await auth.use('api').attempt(body.email, body.password);
    return token;
  }`
```

**Comment by arthurcvm** [comment]
Hi @c4n4r this code is in Controller?

**Reply by c4n4r** [reply]
yes it is

**Comment by thetutlage** [comment, marked as answer]
You can get the types as shown in the docs here.  https://docs.adonisjs.com/guides/context

```ts
import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
```

**Reply by c4n4r** [reply]
Thank you Virk !
